### ISSUE TITLE
Item type for array property 'listFta' is incorrectly named ListFtum

### ISSUE THREAD

**Issue by qgiodau** [joelittlejohn/jsonschema2pojo#1275]
Hi, when i use the following schema:

```
{
  "$schema": "http://json-schema.org/draft-07/schema#",
  "type": "object",
  
  "properties": {
   "foo": {
      "type": "object",      
      "properties": {
        "listFta": {
          "type": "array",
          "items": {
		    "properties": {
              "idScheda": {
                "type": "string",
                "description": "IdScheda"
              }
            }
          }
        }        
      }
    }
  }
}

```

The Foo java class generated, has got a property listFtum instead of listFta:

`
@JsonProperty("listFta")
@Valid
private List<ListFtum> listFta = null;
`

I use the maven plugin for the code generation , but also from website https://www.jsonschema2pojo.org/ , the result is the same.

Can you help me? 

Best regards.
Thank you.

**Comment by unkish** [comment]
Hi

It looks like duplicate of https://github.com/joelittlejohn/jsonschema2pojo/issues/148.
As a workaround You could use the `javaType` extension property to override the class name chosen by the inflector as suggested here: https://github.com/joelittlejohn/jsonschema2pojo/issues/148#issuecomment-34570125
Example:
```{
  "$schema": "http://json-schema.org/draft-07/schema#",
  "type": "object",
  "properties": {
    "foo": {
      "type": "object",
      "properties": {
        "listFta": {
          "type": "array",
          "items": {
            "javaType": "ListFta",
            "properties": {
              "idScheda": {
                "type": "string",
                "description": "IdScheda"
              }
            }
          }
        }
      }
    }
  }
}
```

**Comment by joelittlejohn** [comment]
Yes, so the issue here is that when we see an array like 'Products' we make this singular for the item class name, like 'Product'. For words that end in 'ta' (like data) the singular noun ends in 'tum' (like datum). So the inflector class (the class responsible for producing these singular nouns) is a little over-zealous. 

I don't really like #148 because it encourages a very specific configuration. It would probably be better to add a config option to disable the inflector. As unkish has said though, for now you should be able to work around this using `javaType` or `javaName` inside the items schema.

In the past it has still been useful to add and fix rules in the inflector when we find problems. In this case, it should have a more specific rule I think. The code currently has:

```java
            .singular("([ti])a$", "$1um")
```

which produces these two rules:

* "ia" => "ium"
* "ta" => "tum"

but we should split this like:

```java
            .singular("ia$", "ium")
            .singular("ata$", "atum")
```

so that we get these two rules instead:

* "ia" => "ium"
* "ata" => "atum"

**Comment by qgiodau** [comment]
ok, i will use javaName as suggested.

thank you.

**Comment by unkish** [comment]
> I don't really like #148 because it encourages a very specific configuration

@joelittlejohn would it be possible to elaborate a bit on what is meant by very specific configuration ?

**Comment by eirnym** [comment]
The best implementation I'd see here is make this endings modifications being possible to override by a jsonschema2pojo plugin like for types. In this case everyone can modify default behavior as it suits best to a project

**Comment by joelittlejohn** [comment]
Closed by #1276
